chain/chain/src/chain.rs

```diff
@@ -3824,7 +3908,10 @@ impl Chain {
 
     /// Function to check whether we need to create a new snapshot while processing the current block
     /// Note that this functions is called as a part of block preprocesing, so the head is not updated to current block
-    fn should_make_or_delete_snapshot(&mut self) -> Result<(bool, bool), Error> {
+    fn should_make_or_delete_snapshot(
+        &mut self,
+        next_block: &BlockHeader,
+    ) -> Result<(bool, bool), Error> {
```

Comment:
It seems strange that this method gets the next block as an argument and reads head at the same time. Can you pick one scheme and stick to it? If not feasible can you update the comment?

Reply:
Yeah it is kind of strange… And note that this is done not just in the existing code in this function, but also in the caller. I would guess that you’re right and we could just use block.header().prev_hash() wherever we use the result of this head() call in here. To me seems like it def should always be the same? For now I didn’t touch any of that to avoid changing too many things, but maybe this could be made simpler. For now added a comment to the calling function

Reply:
ok actually, I think this should be fixed... it looks like it's been this way since https://github.com/near/nearcore/pull/9090, where it's calling head() in here when it should be looking at the prev_block field. I think it would make sense to just move this out of start_process_block_impl(), because that doesnt make a lot of sense, like you pointed out in your other comment.

Reply:
removed the block header arg from `Chain::should_make_or_delete_snapshot()` in 6963301b82f4fff65a85222447313c8a3eb7151f. Also refactored the return value a little bit to be more direct